doit/cmdparse.py

```diff
@@ -147,7 +156,16 @@ def help_doc(self):
 
         text = []
         opt_str = self.help_param()
-        opt_help = self.help % {'default': self.default}
+
+        if isinstance(self.type, dict):
+            default = next(iter( key for key in six.iterkeys(self.type)
+                                 if self.type[key] == self.default ))
+            template_dict = {'choices': self.type.keys(),
```

Comment:
lets use `format()` instead of string interpolation `%`.

Since the only value that could appear before was `%(default)s` we can do:

```
if "%(default)s" in self.help:
    # print deprecation warning
    opt_help = self.help % {'default': self.default}
else:
    opt_help = self.help.format(choices=choices, default= self.default) 
```